mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster.go

```diff
@@ -1175,15 +1175,21 @@ func metricsSchema() *schema.Resource {
 	}
 }
 
-func instanceConfigSchema(parent string) *schema.Schema {
-	var instanceConfigKeys = []string{
+// min_num_instances will be picked up only by primary worker instance groups
+func instanceConfigSchemaKeys(parent string) []string {
+	instanceConfigKeys := []string{
 		"cluster_config.0." + parent + ".0.num_instances",
 		"cluster_config.0." + parent + ".0.image_uri",
 		"cluster_config.0." + parent + ".0.machine_type",
-		"cluster_config.0." + parent + ".0.min_cpu_platform",
-		"cluster_config.0." + parent + ".0.disk_config",
 		"cluster_config.0." + parent + ".0.accelerators",
 	}
+	if !strings.Contains(parent, "master") {
+		instanceConfigKeys = append(instanceConfigKeys, "cluster_config.0."+parent+".0.min_num_instances")
+	}
+	return instanceConfigKeys
+}
+
+func instanceConfigSchema(parent string, instanceConfigKeys []string) *schema.Schema {
 
 	masterConfig := strings.Contains(parent, "master")
```

Comment:
Can we inline the new functions and key off of the `masterConfig` boolean here? Right now we're achieving the same result in a few different ways and I'd like to stay consistent.

Reply:
(I'm debating if we want to stop reusing this as a whole, similar to secondary workers. Probably not too much deviation yet, but we're close)